kibana-reports/server/routes/utils/savedSearchReportHelper.ts

```diff
@@ -0,0 +1,217 @@
+  IScopedClusterClient,
+} from '../../../../../src/core/server';
+import { getFileName, callCluster } from './helpers';
+
+/**
+ * Specify how long scroll context should be maintained for scrolled search
+ */
+const scrollTimeout = '1m';
+
+export async function createSavedSearchReport(
+  report: any,
+  client: IClusterClient | IScopedClusterClient
+): Promise<{ timeCreated: number; dataUrl: string; fileName: string }> {
+  const params = report.report_definition.report_params;
+  const reportFormat = params.core_params.report_format;
+  const reportName = params.report_name;
+
+  await populateMetaData(client, report);
+  const data = await generateCsvData(client, params.core_params);
+
+  const curTime = new Date();
+  const timeCreated = curTime.valueOf();
+  const fileName = getFileName(reportName, curTime) + '.' + reportFormat;
+  return {
+    timeCreated,
+    dataUrl: data,
+    fileName,
+  };
+}
+
+/**
+ * Populate parameters and saved search info related to meta data object.
+ * @param client  ES client
+ * @param report  Report input
+ */
+async function populateMetaData(
+  client: IClusterClient | IScopedClusterClient,
+  report: any
+) {
+  metaData.saved_search_id =
+    report.report_definition.report_params.core_params.saved_search_id;
+  metaData.report_format =
+    report.report_definition.report_params.core_params.report_format;
+  metaData.start = report.time_from;
+  metaData.end = report.time_to;
+
+  // Get saved search info
+  let resIndexPattern: any = {};
+  const ssParams = {
+    index: '.kibana',
+    id: 'search:' + metaData.saved_search_id,
+  };
+  const ssInfos = await callCluster(client, 'get', ssParams);
+
+  metaData.sorting = ssInfos._source.search.sort;
+  metaData.type = ssInfos._source.type;
+  metaData.filters =
+    ssInfos._source.search.kibanaSavedObjectMeta.searchSourceJSON;
+
+  // Get the list of selected columns in the saved search.Otherwise select all the fields under the _source
+  await getSelectedFields(ssInfos._source.search.columns);
+
+  // Get index name
+  for (const item of ssInfos._source.references) {
+    if (item.name === JSON.parse(metaData.filters).indexRefName) {
+      // Get index-pattern information
+      const indexPattern = await callCluster(client, 'get', {
+        index: '.kibana',
+        id: 'index-pattern:' + item.id,
+      });
+      resIndexPattern = indexPattern._source['index-pattern'];
+      metaData.paternName = resIndexPattern.title;
+      (metaData.timeFieldName = resIndexPattern.timeFieldName),
+        (metaData.fields = resIndexPattern.fields); // Get all fields
+      // Getting fields of type Date
+      for (const item of JSON.parse(metaData.fields)) {
+        if (item.type === 'date') {
+          metaData.dateFields.push(item.name);
+        }
+      }
+    }
+  }
+}
+
+/**
+ * Generate CSV data by query and convert ES data set.
+ * @param client  ES client
+ * @param limit   limit size of result data set
+ */
+async function generateCsvData(
```

Comment:
np/question: should we rename this `generateSavedSearchData`? I don't see where this is specific to `CSV` format in the method or when it is called in `createSavedSearchReport`

Reply:
forget to rename this. will do. thanks!